sr_robot_commander/src/sr_robot_commander/sr_robot_state_saver.py

```diff
@@ -20,15 +20,30 @@ def __init__(self, name, hand_or_arm="both"):
 
         self.__name = name
 
+        if name is None or name == '':
+            raise Exception("Cannot save with empty name.")
```

Comment:
Using Exception class for custom exceptions is not recommended since it is parent for so many exceptions and catch statement will catch everything. Use specific exception type in this case you could use ValueError https://docs.python.org/2/library/exceptions.html#exceptions.ValueError